File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.js
function solution(participant, completion) {
    const map = new Map();
    
    for (let part of participant) {
        map.set(part, map.has(part) ? map.get(part)+1 : 1);
    }
    for (let com of completion) {
        map.set(com, map.get(com)-1);
    }
    
    for (let [key, value] of map) {
        if (value>0) return key;
    }
}

File: 프로그래머스/1/42576. 완주하지 못한 선수/완주하지 못한 선수.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public String solution(String[] participant, String[] completion) {
        HashMap<String,Integer> hm = new HashMap<>();
        for (int i=0; i<participant.length; i++) {
            String key = participant[i];
            if (hm.containsKey(key))
                hm.put(key, hm.get(key)+1);
            else
                hm.put(key,1);
        }
        for (int i=0; i<completion.length; i++) {
            String key = completion[i];
            hm.put(key, hm.get(key)-1);
        }
        
        String name = "";
        for (Map.Entry<String,Integer> entry: hm.entrySet()) {
            if (entry.getValue()==1) {
                name = entry.getKey();
                break;
            }
        }
        return name;
    }
}